modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/bookmark/BookmarkHelper.kt

```diff
@@ -0,0 +1,117 @@
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+class BookmarkHelper @Inject constructor(
+    private val playbackManager: PlaybackManager,
+    private val bookmarkManager: BookmarkManager,
+    private val settings: Settings,
+) : CoroutineScope {
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Default
+
+    fun handleAddBookmarkAction(
+        context: Context,
+    ) {
+        if (context.isAppForeground()) {
+            val bookmarkIntent =
+                context.packageManager.getLaunchIntentForPackage(context.packageName)
+                    ?.apply { action = INTENT_OPEN_APP_ADD_BOOKMARK }
+            bookmarkIntent?.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            context.startActivity(bookmarkIntent)
+        } else {
+            if (playbackManager.getCurrentEpisode() == null) return
+            launch {
+                val episode = playbackManager.getCurrentEpisode() ?: return@launch
+                val timeInSecs = playbackManager.getCurrentTimeMs(episode) / 1000
+
+                // Load existing bookmark
+                val bookmark = bookmarkManager.findByEpisodeTime(
+                    episode = episode,
+                    timeSecs = timeInSecs
+                )
+
+                if (bookmark == null) {
+                    bookmarkManager.add(
+                        episode = episode,
+                        timeSecs = timeInSecs,
+                        title = context.getString(LR.string.bookmark)
+                    )
+                }
+
+                if (settings.getHeadphoneControlsPlayBookmarkConfirmationSound()) {
+                    playbackManager.playTone()
+                }
+
+                buildAndShowNotification(context)
+            }
+        }
+    }
+}
+
+private fun buildAndShowNotification(
+    context: Context,
+) {
+    val changeTitleAction = NotificationCompat.Action(
+        IR.drawable.ic_notification_edit,
+        context.getString(LR.string.bookmark_notification_action_change_title),
+        buildPendingIntent(context, INTENT_OPEN_APP_ADD_BOOKMARK)
+    )
+
+    val notification = NotificationCompat.Builder(
+        context,
+        Settings.NotificationChannel.NOTIFICATION_CHANNEL_ID_BOOKMARK.id
+    )
+        .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
+        .setContentTitle(context.getString(LR.string.bookmark_notification_title_added))
+        .setContentText(context.getString(LR.string.bookmark_notification_content_tap_to_view))
+        .setSmallIcon(IR.drawable.ic_launcher_foreground)
+        .setAutoCancel(true)
+        .setOnlyAlertOnce(true)
+        .setContentIntent(buildPendingIntent(context, INTENT_OPEN_APP_VIEW_BOOKMARKS))
+        .addAction(changeTitleAction)
```

Comment:
Not for this version but I wonder if we should also have an action for "Delete". It might be a common action they want to take if they accidentally add a bookmark or want to move it. 

![Screenshot_20230811_112853](https://github.com/Automattic/pocket-casts-android/assets/308331/a1485e6b-7952-4069-b3e5-b8aadf3879be)